### PR TITLE
use one shot service

### PR DESCRIPTION
**Pull request by vxav**
This PR:

- Changes static routes setting to use a oneshot service so it persists across reboots.

Tested with `guppy-wc-xav`

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.
- [x] Update Lastpass values if required.

### CODE REVIEW

File: helm/cluster-cloud-director/templates/kubeadmcontrolplane.yaml [hunk 1]
```diff
@@ -112,7 +115,9 @@ spec:
       {{- end }}
     postKubeadmCommands:  
       {{- if $.Values.network.staticRoutes }}
-      {{- include "staticRoutes" . | nindent 6}}
+      - systemctl daemon-reload
+      - systemctl enable static-routes.service
+      - systemctl start static-routes.service
```
Comment (glitchcrab) [inline helm/cluster-cloud-director/templates/kubeadmcontrolplane.yaml:120]:
```suggestion
      - systemctl enable --now static-routes.service
```

entirely up to you if you want to use it, it's just more succinct. (ref: https://documentation.suse.com/smart/linux/html/reference-systemctl-enable-disable-services/index.html#id-1.4)

File: helm/cluster-cloud-director/templates/_helpers.tpl [hunk 2]
```diff
@@ -110,19 +122,24 @@ joinConfiguration:
     kubeletExtraArgs:
       {{- include "kubeletExtraArgs" . | nindent  6}}
       node-labels: "giantswarm.io/node-pool={{ .pool.name }}"
-{{- if $.Values.proxy.enabled }}
 files:
+{{- if $.Values.proxy.enabled }}
```
Comment (glitchcrab) [inline helm/cluster-cloud-director/templates/_helpers.tpl:126]: I'm not sure what moving the `files` key outside of the if statement will do if there are no files to create (no static routes or proxy config) - i don't know if the template will be valid.

File: helm/cluster-cloud-director/templates/kubeadmcontrolplane.yaml (hunk 1 above)
Comment (vxav) [reply helm/cluster-cloud-director/templates/kubeadmcontrolplane.yaml]: Nice, thanks

File: helm/cluster-cloud-director/templates/_helpers.tpl (hunk 2 above)
Comment (glitchcrab) [reply helm/cluster-cloud-director/templates/_helpers.tpl:126]: so it's fine to have an empty `files:` key?

Comment (vxav) [reply helm/cluster-cloud-director/templates/_helpers.tpl:126]: Haven't tested with files but it's fine with other fields (e.g. postkubeadmscript). We'd have to test it in a non-proxied environment to make sure

**Review by glitchcrab — APPROVED** [review]
Assuming it's tested without proxy or routes set and it works then i'm good with it
seal_of_approval

File: helm/cluster-cloud-director/templates/_helpers.tpl [hunk 3]
```diff
@@ -110,19 +122,24 @@ joinConfiguration:
...
 - systemctl daemon-reload
 - systemctl restart containerd
 {{- end }}
 postKubeadmCommands:
 {{- if $.Values.network.staticRoutes }}
-{{- include "staticRoutes" . }}
-{{- end -}}
+- systemctl daemon-reload
+- systemctl enable static-routes.service
+- systemctl start static-routes.service
```
Comment (kopiczko) [inline helm/cluster-cloud-director/templates/_helpers.tpl:141]:
```suggestion
- systemctl enable --now static-routes.service
```

It's weird to do it differently there

**Review by kopiczko — APPROVED** [review]

File: helm/cluster-cloud-director/templates/_helpers.tpl (hunk 3 above)
Comment (glitchcrab) [reply helm/cluster-cloud-director/templates/_helpers.tpl:141]: ah yeah i was lazy and only made the suggestion in one place

Comment (kopiczko) [reply helm/cluster-cloud-director/templates/_helpers.tpl:141]: I missed it was your suggestion.

Comment (vxav) [reply helm/cluster-cloud-director/templates/_helpers.tpl]: https://github.com/giantswarm/cluster-cloud-director/pull/66/commits/5c1dfc044fc4d8b22e39c3cdd4c098b3c860e069

**Comment by vxav** [comment]
> Assuming it's tested without proxy or routes set and it works then i'm good with it seal_of_approval

Tested, worked fine.

**Pull request merged**